src/AutoRest.CSharp/Common/Generation/Writers/JsonCodeWriterExtensions.cs

```diff
@@ -329,13 +330,13 @@ private static void DeserializeIntoObjectProperties(this CodeWriter writer, IEnu
                         var variableOrExpression = writer.DeserializeValue(property.ValueSerialization, $"{itemVariable}.Value");
                         writer.Line($"{propertyVariables[property].Declaration} = {variableOrExpression};");
                     }
-                    else if (property.PropertySerializations is not null)
+                    else if (property.IsFlattenedProperty)
```

Comment:
Well this name collides with something I introduced for the "safe flatten" in mgmt plane.
Shall we rename this to `IsClientFlattenedProperty` to specify?